lib/spack/spack/cmd/install.py

```diff
@@ -31,10 +32,61 @@
 level = "short"
 
 
+def parse_use_buildcache(args):
+    bc_parsed = {}
+    bc_keys = ["package:", "dependencies:", ""]
+    bc_values = ["only", "never", "auto"]
+    kv_list = re.findall("([a-z]+:)?([a-z]+)", args)
+
+    # Verify keys and values
+    bc_map = {k: v for k, v in kv_list if k in bc_keys and v in bc_values}
+    if not len(kv_list) == len(bc_map):
+        unrecognized_args = [(k, v) for k, v in kv_list if k not in bc_keys or v not in bc_values]
+        tty.error("Unrecognized arguments passed to use-buildcache: %s" % unrecognized_args)
```

Comment:
I kind of don't like the way the list of string tuples gets printed, e.g.:

```
$ spack install --use-buildcache always readline
==> Error: Unrecognized arguments passed to use-buildcache: [('', 'always')]
```

What do you think of omitting that bit from this particular line?   Since you print out the expected values below, it seems possible to discern what the problem is.